### PR TITLE
Table: Fix state bug with initialRowIndex

### PR DESCRIPTION
**Pull request by gtk-grafana**
**What is this feature?**
When passing in the `initialRowIndex`, and subsequently passing in an undefined value, the table does not update the state of the `rowHighlightIndex` causing bugs when the calling application no longer wants to highlight a specific row.

Also updating the styles because it's really hard to see the highlighted row in a dark background.

**Why do we need this feature?**
So users can share links to a specific row in the table.

**Who is this feature for?**
Just users of explore logs right now AFAIK

**Special notes for your reviewer:**
Should be pretty simple, and I don't think anyone is using this feature yet besides explore logs, so regression risk should be minimal

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: packages/grafana-ui/src/components/Table/RowsList.tsx [hunk 1]
```diff
@@ -350,7 +353,7 @@ export const RowsList = (props: RowsListProps) => {
       tableState.expanded,
       tableStyles,
       textWrapField,
-      theme.components.table.rowHoverBackground,
+      theme.colors.background.canvas,
```
Comment (nmarrs) [inline packages/grafana-ui/src/components/Table/RowsList.tsx:356]: I'm not sure these current changes are doing anything from my testing locally 🤔 I found though that if we swap out the styles on this line with the `background.canvas` it does have an effect. Either way maybe we should split out the styling changes in a separate PR so @staton-hyse11 (our product designer) can review / not block the bug fix

Current main
Screenshot 2024-07-11 at 8 18 07 PM
Screenshot 2024-07-11 at 8 14 30 PM



Current PR
Screenshot 2024-07-11 at 8 15 20 PM

Screenshot 2024-07-11 at 8 14 39 PM




PR with modification specified
Screenshot 2024-07-11 at 8 18 17 PM
Screenshot 2024-07-11 at 8 14 39 PM

**Review by nmarrs — COMMENTED** [review]
Bug fix looks good - maybe we can separate out styling change into separate PR though so our product designer can take a look at it / not block bug fix 👍

File: packages/grafana-ui/src/components/Table/RowsList.tsx (hunk 1 above)
Comment (gtk-grafana) [reply packages/grafana-ui/src/components/Table/RowsList.tsx]: Yeah, great point, I kinda rushed the original PR adding this functionality and we were left with a background color so subtle that you can't see its selected on screenshares.

Made this PR to deal with that issue separately: https://github.com/grafana/grafana/pull/90364

**Review by nmarrs — APPROVED** [review]
🏓

**Pull request merged**